Add docker file

MaximeBaudette opened the pull request.

dietmarw commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:14.04
```
Why ancient LTS and not the last one 16.04. Having said that there are better Docker friendly variants of Ubuntu available. Have a look at https://github.com/phusion/baseimage-docker I use that for my OM docker

MaximeBaudette replied:
Can I just switch without changing my commands?

dietmarw replied:
Sure. As I said. Look at my setup which you can also test.

MaximeBaudette replied:
ok, I'll look into that

dietmarw commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+                       omniorb-idl \
+	                     git \
+					             python-numpy \
+                       python-omniorb
```
You might wanna switch to Python3 and the new ZeroMQ interface for OMPython. Works more reliable. See also my setup: https://github.com/dietmarw/docker-notebook/blob/master/Dockerfile

MaximeBaudette replied:
I do use ZMQ!

dietmarw replied:
Then you do not need omniorb

dietmarw replied:
the omniorb interface only works for Python 2 whilst the new zeroMQ works for Python 2 and 3. So no reason to install it.

MaximeBaudette replied:
Yep, those lines appear as already removed. (Hence your comment showing as outdated)

dietmarw replied:
I see. Yes I should have looked at the changed files rather then the commit. But then again the commit messages were not very helpful here ;-)

MaximeBaudette replied:
Haha. 
Yes unfortunately I did not manage to dedicate some time to setup everything to locally build docker containers or even test travis script. Also this was my first time dealing with Docker...
And yeah, OMpython and me --> not great friends
Hence, I just pushed and awaited results from dockerhub / travis to see the result. 
Sorry for the mess, this PR is ridiculous I know...

MaximeBaudette commented:
@dietmarw 
Unfortunately I didn't manage to figure out how to switch to the docker image you were recommending, despite changing many things in my files. I'll revert back to how it was, since it was working as intended. 
I'm open to further improvements in another PR

MaximeBaudette commented:
Now I have a working version back. 
If I cannot fix  the previous attempt, let's merge this one.

MaximeBaudette commented:
Now I think I got it alle working following your recommendations @dietmarw

MaximeBaudette commented:
@janlav I'm ready to merge

MaximeBaudette commented:
Unless you wanna do the review @dietmarw ?

dietmarw commented:
Only question I have is if you want python 2 or 3 on the container. I think the default still installs pyhon. Perhaps check that first.

MaximeBaudette commented:
In the build log from docker I get that it installs python 2.7. 
I'm satisfied with that, always have been using this version, and I have never used 3. 

In summary:
- Added the docker file to build a container with latest stable version of OM and OMPython (we had an outdated container and lost the original build instructions)
- Removed the `changeUser.sh` file and made the travis cmds clearer
- Refactored slightly the tests to prepare for regression tests in the future (in case they are implemented some day).
- Improved the report to make it easier to read in Travis
- Bonus, this morning I thought about testing the CI scripts, and I implemented a first testing stage. 

I feel like it's enough for this PR.
